python/desc/sims/GCRCatSimInterface/TruthCatalogLC.py

```diff
@@ -0,0 +1,333 @@
+
+    ra_colname is the column used for RA of the pointing (default:
+    descDitheredRA)
+
+    dec_colname is the column used for the Dec of the pointing (default:
+    descDitheredDec)
+
+    sql_file_name is the name of the parameter database produced by
+    write_sprinkled_param_db to be used
+
+    bp_dict is a BandpassDict of the telescope filters to be used
+
+    Returns
+    -------
+    None
+
+    Writes out a database to out_file_name.  The tables of this database and
+    their columns are:
+
+    agn_lc/sn_lc:
+        - uniqueId -- an int unique to all objects
+        - obshistid -- an int unique to all pointings
+        - mag -- the magnitude observed for this object at that pointing
+
+    obs_metadata:
+        - obshistid -- an int unique to all pointings
+        - mjd -- the date of the pointing
+        - filter -- an int corresponding to the telescope filter (0==u, 1==g..)
+
+    sprinkled_objects:
+        - uniqueId -- an int unique to all objects
+        - galaxy_id -- an int indicating the host galaxy
+        - ra -- in degrees
+        - dec -- in degrees
+    """
+
+    t0_master = time.time()
+
+    if not os.path.isfile(sql_file_name):
+        raise RuntimeError('%s does not exist' % sql_file_name)
+
+
+    sn_simulator = SneSimulator(bp_dict)
+    sed_dir = os.environ['SIMS_SED_LIBRARY_DIR']
+
+    create_sprinkled_sql_file(out_file_name)
+
+    t_start = time.time()
+    (htmid_dict,
+     mjd_dict,
+     filter_dict) = get_pointing_htmid(pointing_dir, opsim_db_name,
+                                       ra_colname=ra_colname,
+                                       dec_colname=dec_colname)
+    t_htmid_dict = time.time()-t_start
+
+    with sqlite3.connect(out_file_name) as conn:
+        cursor = conn.cursor()
+        values = ((obs, mjd_dict[obs], filter_dict[obs])
+                  for obs in mjd_dict)
+        cursor.executemany('''INSERT INTO obs_metadata VALUES (?,?,?)''', values)
+        cursor.execute('''CREATE INDEX obs_filter
+                       ON obs_metadata (obshistid, filter)''')
+        conn.commit()
+
+    print('\ngot htmid_dict -- %d in %e seconds' % (len(htmid_dict), t_htmid_dict))
+
+    db = DBObject(sql_file_name, driver='sqlite')
+
+    query = 'SELECT DISTINCT htmid FROM zpoint WHERE is_agn=1'
+    dtype = np.dtype([('htmid', int)])
+
+    results = db.execute_arbitrary(query, dtype=dtype)
+
+    object_htmid = results['htmid']
+
+    agn_dtype = np.dtype([('uniqueId', int), ('galaxy_id', int),
+                          ('ra', float), ('dec', float),
+                          ('redshift', float), ('sed', str, 500),
+                          ('magnorm', float), ('varParamStr', str, 500)])
+
+    agn_base_query = 'SELECT uniqueId, galaxy_id, '
+    agn_base_query += 'raJ2000, decJ2000, '
+    agn_base_query += 'redshift, sedFilepath, '
+    agn_base_query += 'magNorm, varParamStr '
+    agn_base_query += 'FROM zpoint WHERE is_agn=1 '
```

Comment:
Either that, or I need to rename the table.  I may have originally intended this table to just contain sprinkled objects, but now I think it should contain light curves for all AGN and SNe, sprinkled or otherwise.

Do you think there is value in keeping the sprinkled light curves separate from the unsprinkled light curves, or will people be able to discern them based on position relative to the DDF?

Reply:
What about adding the `is_sprinkled` column and then putting them all in one table? That would make it easy to search within sqlite for just the sprinkled objects.

Reply:
Haha, I just noticed that's what you ended up doing. Maybe I should go through the diff first...